src/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1486,5 +1499,112 @@ private bool TryGetHostUri(string hostName, out Uri hostUri)
+                handler.PreAuthenticate = parameters.PreAuthenticate;
+                client.Timeout = parameters.Timeout;
+
+                if (parameters.CookieContainer != null)
+                {
+                    handler.CookieContainer = parameters.CookieContainer;
+                    Debug.Assert(handler.UseCookies); // Default of handler.UseCookies is true.
+                }
+                else
+                {
+                    handler.UseCookies = false;
+                }
+
+                Debug.Assert(handler.UseProxy); // Default of handler.UseProxy is true.
+                Debug.Assert(handler.Proxy == null); // Default of handler.Proxy is null.
+
+                // HttpClientHandler default is to use a proxy which is the system proxy.
+                // This is indicated by the properties 'UseProxy == true' and 'Proxy == null'.
+                //
+                // However, HttpWebRequest doesn't have a separate 'UseProxy' property. Instead,
+                // the default of the 'Proxy' property is a non-null IWebProxy object which is the
+                // system default proxy object. If the 'Proxy' property were actually null, then
+                // that means don't use any proxy.
+                //
+                // So, we need to map the desired HttpWebRequest proxy settings to equivalent
+                // HttpClientHandler settings.
+                if (parameters.Proxy == null)
+                {
+                    handler.UseProxy = false;
+                }
+                else if (!object.ReferenceEquals(parameters.Proxy, WebRequest.GetSystemWebProxy()))
+                {
+                    handler.Proxy = parameters.Proxy;
+                }
+                else
+                {
+                    // Since this HttpWebRequest is using the default system proxy, we need to
+                    // pass any proxy credentials that the developer might have set via the
+                    // WebRequest.DefaultWebProxy.Credentials property.
+                    handler.DefaultProxyCredentials = parameters.Proxy.Credentials;
+                }
+
+                if (parameters.ClientCertificates != null)
+                {
+                    handler.ClientCertificates.AddRange(parameters.ClientCertificates);
+                }
+
+                // Set relevant properties from ServicePointManager
+                handler.SslProtocols = (SslProtocols)ServicePointManager.SecurityProtocol;
+                handler.CheckCertificateRevocationList = ServicePointManager.CheckCertificateRevocationList;
+                RemoteCertificateValidationCallback rcvc = parameters.ServerCertificateValidationCallback;
+                if (rcvc != null)
+                {
+                    RemoteCertificateValidationCallback localRcvc = rcvc;
+                    handler.ServerCertificateCustomValidationCallback = (message, cert, chain, errors) => localRcvc(request, cert, chain, errors);
```

Comment:
I'm not understanding the logic around `request` being passed in to CreateHttpClient.  `null` is passed in when creating the cached instance, and the instance is passed in otherwise.  That means then that for the cached instance, this `request` will be null.  And that means that anyone using the cached HttpClient instance will have a null request object passed into their RemoteCertificateValidationCallback  Seems like a bug.

Reply:
Also, there's a subtle change here that's going to result in the closure being created even when the ServerCertificateValidationCallback is null.  Previously the only state being closed over were locals, this, or confined to the immediate containing scope.  This change now also closes over `request`, which has method scope.  That means the closure for this lambda would previously have only been allocated with rcvc != null, but now it's going to be allocated every time this method is called.

Reply:
1. Passing null is intentional because client caching is only allowed when ServiceCertificateValidationCallback == null, thus it will never be caught in the lambda. Additionally, even if somebody accidentally change caching conditions to allow validation callback caching, passing null will still protect from leaking of "this" instance in the static cache.

2. Valid point. I will fix it.